lib/rspec/mocks/space.rb

```diff
@@ -164,6 +171,24 @@ def proxy_not_found_for(id, object)
                       end
       end
 
+      def superclass_proxy_not_found_for(id, object)
+        raise "superclass_proxy_not_found_for called with something that is not a class" unless Class === object
+        proxies[id] = class_proxy_with_callback_verification_strategy(object, NoCallbackInvocationStrategy.new)
+      end
+
+      def class_proxy_with_callback_verification_strategy(object, strategy)
+        if RSpec::Mocks.configuration.verify_partial_doubles?
+          VerifyingPartialClassDoubleProxy.new(
+            self,
+            object,
+            @expectation_ordering,
+            strategy
+          )
+        else
+          PartialClassDoubleProxy.new(self, object, @expectation_ordering)
+        end
+      end
```

Comment:
I was hoping to avoid needing to do the lookup / check on every call. Instead we just send the message to an adapter. However, my main goal, was simplifying this part of the `Space` code.

How the configuration manages the adapter I view as a "private implementation detail". I don't have too strong an opinion on how that happens.